compiler/src/Language/Bond/Codegen/Java/Enum_java.hs

```diff
@@ -30,37 +30,73 @@ package #{javaPackage};
 
     typeDefinition Enum {..} = [lt|
 #{Java.generatedClassAnnotations}
-public class #{declName} {
-    #{newlineSep 1 constant enumConstantsWithInt}
+public final class #{declName} {
+
+    public static final class Values {
+        private Values() {}
+
+        #{newlineSep 2 constantIntValueDecl enumConstantsWithInt}
+    }
+
+    #{newlineSep 1 constantObjectDecl enumConstantsWithInt}
 
     public final int value;
 
     public #{declName}(int value) { this.value = value; }
 
     @Override
-    public boolean equals(Object other) {
-        if (!(other instanceof #{declName})) { return false; }
-        return this.value == ((#{declName}) other).value;
-    }
+    public boolean equals(Object other) { return (other instanceof #{declName}) && (this.value == ((#{declName}) other).value); }
+ 
+    @Override
+    public int hashCode() { return this.value; }
 
     @Override
-    public int hashCode() {
-        return this.value;
+    public String toString() { return String.valueOf(this.value); }
+
+    public static #{declName} get(int value) {
+        switch (value) {
+            #{newlineSep 3 switchCaseConstantMapping enumConstantsWithIntDistinct}
```

Comment:
The use of the `Distinct` flavor is to avoid two case labels for ints with the same value, right? Is it a compile error to have two equivalent labels? My gut says to emit both labels for human clarity, let the compiler ignore the redundant ones, and let the `@generated` annotation suppress the warning I would think this causes.